test/inductor/test_cpu_select_algorithm.py

```diff
@@ -2069,6 +2140,83 @@ def forward(self, x, other):
         self.assertEqual(counters["inductor"]["select_algorithm_autotune"], 1)
         self.assertEqual(counters["inductor"]["cpp_epilogue_fusion_counter"], 1)
 
+    # We need it for the per-tensor case because Dynamo will end up specializing batch dimension,
+    # which we marked as dynamic with mark_dynamic
+    # This might be due to a bug in Dynamo
+    @dynamo_config.patch({"allow_ignore_mark_dynamic": True})
```

Comment:
Hi @leslie-fang-intel, I used this config because when per-tensor quantization of activation was used, there was a runtime error with Dynamo guards because Dynamo ended up specializing the batch dimension.

Could this be due to some bug in Dynamo?



The precise error in logs taken with `TORCH_LOGS="+dynamic"` is:
```
E1215 22:23:18.202000 2283097 torch/_guards.py:295] [0/0] Error while creating guard:
E1215 22:23:18.202000 2283097 torch/_guards.py:295] [0/0] Name: ''
E1215 22:23:18.202000 2283097 torch/_guards.py:295] [0/0]     Source: shape_env
E1215 22:23:18.202000 2283097 torch/_guards.py:295] [0/0]     Create Function: SHAPE_ENV
E1215 22:23:18.202000 2283097 torch/_guards.py:295] [0/0]     Guard Types: None
E1215 22:23:18.202000 2283097 torch/_guards.py:295] [0/0]     Code List: None
E1215 22:23:18.202000 2283097 torch/_guards.py:295] [0/0]     Object Weakref: None
E1215 22:23:18.202000 2283097 torch/_guards.py:295] [0/0]     Guarded Class Weakref: None
E1215 22:23:18.202000 2283097 torch/_guards.py:295] [0/0] Traceback (most recent call last):
E1215 22:23:18.202000 2283097 torch/_guards.py:295] [0/0]   File "torch/_guards.py", line 293, in create
E1215 22:23:18.202000 2283097 torch/_guards.py:295] [0/0]     return self.create_fn(builder, self)
E1215 22:23:18.202000 2283097 torch/_guards.py:295] [0/0]   File "torch/_dynamo/guards.py", line 1863, in SHAPE_ENV
E1215 22:23:18.202000 2283097 torch/_guards.py:295] [0/0]     code_parts, verbose_code_parts = output_graph.shape_env.produce_guards_verbose(
E1215 22:23:18.202000 2283097 torch/_guards.py:295] [0/0]   File "torch/fx/experimental/symbolic_shapes.py", line 5190, in produce_guards_verbose
E1215 22:23:18.202000 2283097 torch/_guards.py:295] [0/0]     raise ConstraintViolationError(
E1215 22:23:18.202000 2283097 torch/_guards.py:295] [0/0] torch.fx.experimental.symbolic_shapes.ConstraintViolationError: Constraints violated (L['ex'][0].size()[0])! For more information, run with TORCH_LOGS="+dynamic".
E1215 22:23:18.202000 2283097 torch/_guards.py:295] [0/0]   - Not all values of RelaxedUnspecConstraint(L['ex'][0].size()[0]) are valid because L['ex'][0].size()[0] was inferred to be a constant (32).
E1215 22:23:18.202000 2283097 torch/_guards.py:297] [0/0] Created at:
E1215 22:23:18.202000 2283097 torch/_guards.py:297] [0/0]   File "torch/_dynamo/convert_frame.py", line 642, in transform
E1215 22:23:18.202000 2283097 torch/_guards.py:297] [0/0]     tracer = InstructionTranslator(
E1215 22:23:18.202000 2283097 torch/_guards.py:297] [0/0]   File "torch/_dynamo/symbolic_convert.py", line 2709, in __init__
E1215 22:23:18.202000 2283097 torch/_guards.py:297] [0/0]     output=OutputGraph(
E1215 22:23:18.202000 2283097 torch/_guards.py:297] [0/0]   File "torch/_dynamo/output_graph.py", line 336, in __init__
E1215 22:23:18.202000 2283097 torch/_guards.py:297] [0/0]     self.init_ambient_guards()
E1215 22:23:18.202000 2283097 torch/_guards.py:297] [0/0]   File "torch/_dynamo/output_graph.py", line 485, in init_ambient_guards
E1215 22:23:18.202000 2283097 torch/_guards.py:297] [0/0]     self.guards.add(ShapeEnvSource().make_guard(GuardBuilder.SHAPE_ENV))
```

```
Not all values of RelaxedUnspecConstraint(L['ex'][0].size()[0]) are valid because L['ex'][0].size()[0] was inferred to be a constant (32).
```

<img width="1919" alt="image" src="https://github.com/user-attachments/assets/b13936d8-9a8d-4d34-b973-1126df8adaa1" />

Reply:
@leslie-fang-intel, due to #135960, it looks like you had encountered this error in the past. Please advise on the best course-of-action. Thanks!

Reply:
@leslie-fang-intel, thanks, your intuition was right. this error went away after changing `K` to a different value than `M`.